### PR TITLE
Contribute a PR template

### PR DESCRIPTION
**Pull request by devoncarew**
- contribute a PR template
- update the main CONTRIBUTING.md file

Screenshot 2023-05-08 at 10 38 44 AM

### CODE REVIEW

**Review by natebosch — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,6 +31,42 @@ The Dart source code in this repo follows the:
...
+- changes to tests
+
+And generally, the pubspec version and the changelog version should always
+agree.
+
+## Testing
+
+Changes to dart-lang repos must have tests, except for PRs that:
+
+- only affect comments (including documentation)
```
Comment (mosuem) [inline CONTRIBUTING.md:61]: Maybe rather say something along the lines of 
```
Changes to dart-lang repos which modify production code must have tests, except for PRs that 
- are generated by automated bots (rollers)
- have an explicit exemption from a repository contributor
```
? But this is probably bikeshedding.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -31,6 +31,42 @@ The Dart source code in this repo follows the:
...
+for a description of the process.
+
+## Changelog entries
+
+Most changes should be accompanied by a new entry in the CHANGELOG.md file. A
+good rule of thumb is that if the change is user-facing, it should have a
+changelog entry. This includes changes to documentation - like a README.md file
+update - or changes to dartdoc documentation.
+
+Some changes which are exempt are:
```
Comment (mosuem) [inline CONTRIBUTING.md:49]: Why explicitly name these exceptions? Do they not follow from the description above? Also, `some changes` leaves ambiguity, so I would rather just not talk about exceptions and leave that to common sense.

**Review by mosuem — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (devoncarew) [reply CONTRIBUTING.md:61]: Hmm, I think I'd like to keep this list as-is (it's borrowed from the flutter PR template, and I think is pretty reasonable). But happy to update this section on an on-going basis.

I did adjust the final sentence to 'Repository contributors may indicate that a PR should have a test, even for the general exemptions above.'.

**Comment by devoncarew** [comment]
The updated PR template text:

Screenshot 2023-05-09 at 9 36 53 AM

**Pull request merged**